libtenzir/builtins/formats/pcap.cpp

```diff
@@ -437,21 +436,19 @@ auto to_packet_record(auto row) -> std::pair<packet_record, uint32_t> {
       pkt.header.captured_packet_length = caf::get<uint64_t>(value);
     else if (key == "original_packet_length")
       pkt.header.original_packet_length = caf::get<uint64_t>(value);
-    else if (key == "data")
-      pkt_data = caf::get<std::string_view>(value);
-    else
+    else if (key == "data") {
+      pkt.data = caf::get<view<blob>>(value);
```

Comment:
The potential type confusion here can now easily be run into when users store packets and write them here.

I think we basically must use `get_if` here already.

Reply:
I agree (see https://github.com/tenzir/issues/issues/946). I can fix this with a fallback to `string` for this special case and ignoring the data otherwise, and would leave deeper refactorings for a future PR.

Reply:
The fallback makes sense for now.